src/main/kotlin/no/fdk/dataset_catalog/service/DatasetService.kt

```diff
@@ -29,7 +30,7 @@ class DatasetService (
     fun create(catalogId: String, dataset: Dataset) {
         dataset
             .copy(
-                id = UUID.randomUUID().toString(),
+                id = uuidUtil.randomUUID(),
```

Comment:
why?

Reply:
for å kunne mocke det?

Reply:
I så fall kommer du til å få akkurat samme problem i kontraktstestene, og da kan du ikke mocke :P

Bedre å bare håndtere det i test. Sjekk at de har fått id så endrer du den til en test-id og kjører equals.